xdsl/ir/core.py

```diff
@@ -790,13 +790,20 @@ def verify(self, verify_nested_ops: bool = True) -> None:
                     )
 
         if parent_block is not None and parent_region is not None:
-            # TODO single-block regions dealt when the NoTerminator trait is
-            # implemented (https://github.com/xdslproject/xdsl/issues/1093)
-            if len(parent_region.blocks) > 1:
-                if parent_block.last_op is self and not self.has_trait(IsTerminator):
-                    raise VerifyException(
-                        f"Operation {self.name} terminates block in multi-block region but is not a terminator"
-                    )
+            if parent_block.last_op == self:
+                if len(parent_region.blocks) == 1:
+                    if (
```

Comment:
Is there a case missing where we have `NoTerminator` but have been given a terminator?
Or is this handle somewhere else?

Reply:
do you mean `Block.verify()`?